jaeger_client/metrics.py

```diff
@@ -21,69 +21,115 @@
+        a callable function used to record a float duration in microseconds.
+        :param name: name of the timer
+        :param tags: tags for the timer
+        :return: a callable function which takes the duration to
+        record ie. def record(duration)
+        """
+        return self._noop
+
+    def create_gauge(self, name, tags=None):
+        """
+        Generates a new gauge from the given name and tags and returns
+        a callable function used to update the gauge.
+        :param name: name of the gauge
+        :param tags: tags for the gauge
+        :return: a callable function which takes the value to update
+        the gauge with ie. def update(value)
+        """
+        return self._noop
 
-    TRACES_JOINED_SAMPLED = '%s.traces-joined.sampled' % prefix
-    """Number of externally started sampled traces this tracer joined"""
 
-    TRACES_JOINED_NOT_SAMPLED = '%s.traces-joined.not-sampled' % prefix
-    """Number of externally started non-sampled traces this tracer joined"""
+class LegacyMetricsFactory(MetricsFactory):
+    """A MetricsFactory adapter for legacy Metrics class."""
 
-    SPANS_SAMPLED = '%s.spans.sampled' % prefix
-    """Number of sampled spans started/finished by this tracer"""
+    def __init__(self, metrics):
+        self._metrics = metrics
 
-    SPANS_NOT_SAMPLED = '%s.spans.not-sampled' % prefix
-    """Number of not-sampled spans started/finished by this tracer"""
+    def create_counter(self, name, tags=None):
+        key = self._get_key(name, tags)
 
-    TRACER_DECODING_ERRORS = '%s.decoding.errors' % prefix
-    """Number of errors decoding tracing context"""
+        def increment(value):
+            return self._metrics.count(key, value)
+        return increment
 
-    REPORTER_SUCCESS = '%s.spans.reported' % prefix
-    """Number of spans successfully reported"""
+    def create_timer(self, name, tags=None):
+        key = self._get_key(name, tags)
 
-    REPORTER_FAILURE = '%s.spans.failed' % prefix
-    """Number of spans in failed attempts to report"""
+        def record(value):
+            # Convert microseconds to milliseconds for legacy
+            return self._metrics.timing(key, value / 1000.0)
+        return record
 
-    REPORTER_DROPPED = '%s.spans.dropped' % prefix
-    """Number of spans dropped from the reporter"""
+    def create_gauge(self, name, tags=None):
+        key = self._get_key(name, tags)
 
-    REPORTER_SOCKET = '%s.spans.socket_error' % prefix
-    """Number of spans dropped due to socket error"""
+        def update(value):
+            return self._metrics.gauge(key, value)
+        return update
 
-    REPORTER_QUEUE_LENGTH = '%s.queue_length' % prefix
-    """Current number of spans in the reporter queue"""
+    def _get_key(self, name, tags=None):
+        if not tags:
+            return name
+        key = name
+        for k in sorted(tags.iterkeys()):
+            key = key + '.' + str(k) + '_' + str(tags[k])
```

Comment:
we need to normalize the values to a set of strings that are safe for metric name, because operation names can be anything. See how it's done in the Go client (doesn't have to be as flexible as there). And let's add tests to check that operation names are properly sanitized.